main.go

```diff
@@ -58,6 +59,15 @@ func main() {
 	app.Email = "https://github.com/docker/machine"
 	app.Before = func(c *cli.Context) error {
 		os.Setenv("MACHINE_STORAGE_PATH", c.GlobalString("storage-path"))
+		switch c.GlobalString("ssh-client-type") {
+		case "gonative":
+			ssh.SetDefaultClient(ssh.GoNative)
+		case "shellout":
+			ssh.SetDefaultClient(ssh.Shellout)
+		case "":
```

Comment:
Seems like this whole bit should just checking for "gonative" and using shellout option otherwise.